src/bug-filing/common/issue-filing-url-string-utils.ts

```diff
@@ -4,36 +4,35 @@ import { EnvironmentInfo } from '../../common/environment-info-provider';
 import { title } from '../../content/strings/application';
 import { CreateIssueDetailsTextData } from './../../common/types/create-issue-details-text-data';
 
-export class IssueFilingUrlStringUtils {
-    public static footer(environmentInfo: EnvironmentInfo): string {
+export type IssueUrlCreationUtils = {
+    footer: (info: EnvironmentInfo) => string;
+    collapseConsecutiveSpaces: (input: string) => string;
+    markdownEscapeBlock: (input: string) => string;
+    getSelectorLastPart: (selector: string) => string;
+    standardizeTags: (data: CreateIssueDetailsTextData) => string[];
+};
+
+export const IssueFilingUrlStringUtils: IssueUrlCreationUtils = {
+    footer: (environmentInfo: EnvironmentInfo): string => {
         return (
             `This accessibility issue was found using ${title} ` +
             `${environmentInfo.extensionVersion} (axe-core ${environmentInfo.axeCoreVersion}), ` +
             'a tool that helps find and fix accessibility issues. Get more information & download ' +
             'this tool at http://aka.ms/AccessibilityInsights.'
         );
-    }
-
-    public static collapseConsecutiveSpaces(input: string): string {
-        return input.replace(/\s+/g, ' ');
-    }
-
-    public static markdownEscapeBlock(input: string): string {
-        return input
+    },
+    collapseConsecutiveSpaces: (input: string): string => input.replace(/\s+/g, ' '),
+    markdownEscapeBlock: (input: string): string =>
```

Comment:
"escape" strongly implies that you would be sanitizing any characters in `input` that markdown would treat specially by replacing them with an escape sequence. This method would be better named something like `encloseInMarkdownCodeBlock` or `formatAsMarkdownCodeBlock`.

I know you didn't introduce this name in your PR, but misusing the term "escape" is so likely to result in a security vulnerability that I want to point it out anyway while I'm noticing it. We need to rename this and audit all of the use sites for any that are expecting it to actually result in the sanitization of untrusted input.

Reply:
Rename!
We're applying this function to data from we get from axe (`failureSummary`). Not sure if we need to actually escape this or not. We're using this info to create a url to create the issue/bug against github/azure dev ops so basically we're transferring this risk to a 3rd party. If you still want this to be done, I'm fine doing it on a separated PR (as this one is getting old and I would really like to getting completed soon)